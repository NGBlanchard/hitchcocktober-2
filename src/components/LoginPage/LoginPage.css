* {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  font-family: "Raleway", sans-serif;
  font-size: 17px;
}

.LoginPage {
  margin-left: 20px;
  margin-right: 20px;
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
  justify-content: center;
  color: rgb(69, 69, 69);
}

h2 {
  margin-top: 150px;
  margin-bottom: 20px;
}



.red {
  color: #ca1551;
}

.success-alert {
  display: none;
}


.Input,
.Textarea {
  border: 1px solid #ddd;
  display: block;
  font-size: 1.1em;
  margin-bottom: 10px;
  padding: 10px;
  text-align: left;
  width: 100%;
  border-radius: 4px;
  color: rgb(69, 69, 69);
}
textarea:focus,
input:focus {
  outline: none;
}

.Input::placeholder,
.Textarea::placeholder {
  color: #ccc;
}

.Textarea {
  resize: vertical;
}

.welcome-header {
  height: 120px;
  color: rgb(69, 69, 69);
  text-align: center;
  margin-top: 10px;
  margin-bottom: 50px;
}

