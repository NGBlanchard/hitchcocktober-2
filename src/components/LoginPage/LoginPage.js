import React, { Component } from "react";
import LoginForm from "../../components/LoginForm/LoginForm";
import LoginNav from "../Nav/LoginNav";
import Context from "../../Context";
import "./LoginPage.css";
import config from "../../config";
import TokenService from "../../services/token-service";

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  };
  static contextType = Context;

  getUserData(id) {
    return fetch(`${config.API_ENDPOINT}/users/${id}`, {
      headers: {
        authorization: `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res =>
        !res.ok ? res.json().then(e => Promise.reject(e)) : res.json()
      )
      .then(res => this.context.setBigObj(res));
  }

  getMovies() {
    return fetch(`${config.API_ENDPOINT}/movies`, {
      headers: {}
    })
      .then(res =>
        !res.ok ? res.json().then(e => Promise.reject(e)) : res.json()
      )
      .then(res => this.context.setList(res));
  }

  getUserBigObject(id) {
    return Promise.all([this.getMovies(), this.getUserData(id)]);
  }

  onLoginSuccess = id => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || "/calendar";
    this.getUserBigObject(id).then(([movies, userData]) => {
      history.push(destination);
    });
  };

  render() {
    return (
      <>
        <LoginNav />
        <section className="LoginPage">
          <div className="success-alert"></div>
          <h2>Welcome to Hitchcocktober</h2>
          <p>Please login below</p>
          <LoginForm onLoginSuccess={this.onLoginSuccess} />
        </section>
      </>
    );
  }
}
