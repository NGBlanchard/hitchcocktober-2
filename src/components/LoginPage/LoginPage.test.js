import LoginPage from "./LoginPage";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import toJson from "enzyme-to-json";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<LoginPage />", () => {
  it("Renders without crashing", () => {
    mount(
      <Router>
        <LoginPage />
      </Router>
    );
  });
  it("renders Login Page by default", () => {
    const wrapper = mount(
      <Router>
        <LoginPage />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
