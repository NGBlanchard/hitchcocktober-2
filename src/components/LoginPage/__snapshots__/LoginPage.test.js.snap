// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders Login Page by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LoginPage
      history={
        Object {
          "push": [Function],
        }
      }
      location={Object {}}
    >
      <LoginNav>
        <nav
          className="Nav"
        >
          <Link
            to="/list"
          >
            <LinkAnchor
              href="/list"
              navigate={[Function]}
            >
              <a
                href="/list"
                onClick={[Function]}
              >
                Movies
              </a>
            </LinkAnchor>
          </Link>
           
          <Link
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Register
              </a>
            </LinkAnchor>
          </Link>
          <div
            className="Nav__not-logged-in"
          >
            <Link
              to="/login"
            >
              <LinkAnchor
                href="/login"
                navigate={[Function]}
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  Log in
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </nav>
      </LoginNav>
      <section
        className="LoginPage"
      >
        <div
          className="success-alert"
        />
        <h1
          className="welcome-header"
        >
          <img
            alt="welcome"
            className="welcome-header"
            src="welcome.png"
          />
        </h1>
        <p>
          Please login below
        </p>
        <LoginForm
          onLoginSuccess={[Function]}
        >
          <form
            className="LoginForm"
            onSubmit={[Function]}
          >
            <span
              role="alert"
            />
            <section
              className="user_name"
            >
              <label
                htmlFor="LoginForm__user_name"
              >
                Username
              </label>
              <input
                id="LoginForm__user_name"
                name="user_name"
              />
            </section>
            <section
              className="password"
            >
              <label
                htmlFor="LoginForm__password"
              >
                Password
              </label>
              <input
                id="LoginForm__password"
                name="password"
                type="password"
              />
            </section>
            <button
              type="submit"
            >
              Login
            </button>
          </form>
        </LoginForm>
      </section>
    </LoginPage>
  </Router>
</BrowserRouter>
`;
